### DISCUSSION TITLE
oncrash signal?

### DISCUSSION THREAD

**Discussion by hongkongkiwi** [finit-project/finit#245] [General]
Is it possible to add a condition that’s automatically set when a service crashes? e.g <service/myservice/crashed> And additionally <service/myservice/running> ?

I’ve been testing respawn feature but I find I would like to handle crashes and attempt to fix the situation in some cases (E.g restore default mosquito config).

I think this feature could mean you don’t need to extend oncrash: later because a task (or event) can be setup to handle the case of a crash and run a script (or reboot) via the built in signal mechanism

**Comment by troglobit** [comment]
We recently gained support for run/tasks to assert conditions for their return status, e.g., `task/foo/failure` and `task/foo/success`.  Shouldn't be too much work adding this also for services.

**Reply by troglobit** [reply]
I've created issue #246 for this.
